guides/cheatsheets/axon_pytorch.cheatmd

```diff
@@ -0,0 +1,546 @@
+Axon.dropout(rate: 0.5, name: "dropout1")
+
+# Usage is implicit within the model's structure.
+# The mode (:train or :infer) is passed to the model execution function.
+# {init_fn, predict_fn} = Axon.build(model, mode: :train)
+# predict_fn.(params, inputs, mode: :train)
+```
+
+#### PyTorch
+
+Applies dropout during training (`model.train()` mode). It's a no-op during evaluation (`model.eval()` mode).
+
+```python
+# p is the probability of an element being zeroed.
+dropout1 = nn.Dropout(p=0.5)
+
+# Example usage:
+# model.train() # Set model to training mode
+# output = dropout1(x)
+
+# model.eval() # Set model to evaluation mode (dropout becomes identity)
+# output_eval = dropout1(x) # dropout1 has no effect here
+```
+
+### Normalization (LayerNorm)
+
+Applies Layer Normalization over a mini-batch of inputs.
+Normalizes the activations of the previous layer for each given example independently.
+
+#### Axon
+```elixir
+# Typically applied to the feature dimension(s).
+Axon.layer_norm(name: "layernorm1")
+
+# Can specify the axis/axes for normalization (default is usually the last axis)
+# Axon.layer_norm(axis: -1, epsilon: 1.0e-5)
+```
+
+#### PyTorch
+```python
+# Provide the shape of the features to normalize over.
+# This typically means the last dimension(s) of the tensor.
+# Example 1: Input (N, features_dim), normalize over features_dim
+# features_dim = 128
+# layernorm1 = nn.LayerNorm(features_dim)
+
+# Example 2: Input (N, C, H, W), normalize over C, H, W:
+# normalized_shape = [C, H, W] # Needs actual channel, height, width values
+# layernorm2 = nn.LayerNorm(normalized_shape)
+
+# Example 3: Common case in Transformers (Input: N, SeqLen, EmbedDim):
+embed_dim = 512
+layernorm_transformers = nn.LayerNorm(embed_dim)
+
+# Example usage (assuming input `x` with shape (N, SeqLen, embed_dim)):
+# output = layernorm_transformers(x)
+```
+
+## Activation Functions
+
+#### Axon
+
+Activations are typically specified as options within layers (like `Axon.dense`) or applied as separate layers in the model definition pipeline.
+
+```elixir
+# Option 1: As layer option
+model = Axon.input("input", shape: {nil, 784})
+        |> Axon.dense(128, activation: :relu)
+
+# Option 2: As separate layer
+model = Axon.input("input", shape: {nil, 10})
+        |> Axon.softmax()
+
+# Common activation atoms: :relu, :softmax, :sigmoid, :tanh, :identity, etc.
+# Custom functions can also be used.
+```
+
+#### PyTorch
+
+PyTorch also supports a variety of activation functions, including built-in ones and custom implementations.
+
+```python
+relu = nn.ReLU()
+softmax = nn.Softmax(dim=1)
+sigmoid = nn.Sigmoid()
+tanh = nn.Tanh()
+
+output = relu(x)
+output = softmax(x)
+output = sigmoid(x)
+output = tanh(x)
+```
+
+## Defining Custom Layers/Models
+
+#### Axon
+
+Axon allows for the definition of custom layers and models.
+
+```elixir
+# Example:
+defmodule MyCustomLayer do
+  def call(x) do
+    Axon.dense(128, activation: :relu, name: "my_dense_layer")
```

Comment:
```suggestion
    Axon.dense(x, 128, activation: :relu, name: "my_dense_layer")
```

Reply:
This would be a good section to mention `Axon.block`, which is kind of like an `nn.Module` in that it creates a re-usable block